core/modules/update_ttl.cc

```diff
@@ -0,0 +1,44 @@
+#include "update_ttl.h"
+#include "../utils/ether.h"
+#include "../utils/ip.h"
+
+using bess::utils::EthHeader;
+using bess::utils::Ipv4Header;
+
+void UpdateTTL::ProcessBatch(bess::PacketBatch *batch) {
+  bess::PacketBatch out_batch;
+  bess::PacketBatch free_batch;
+  out_batch.clear();
+  free_batch.clear();
+
+  int cnt = batch->cnt();
+
+  for (int i = 0; i < cnt; i++) {
+    bess::Packet *pkt = batch->pkts()[i];
+
+    struct EthHeader *eth = pkt->head_data<struct EthHeader *>();
+    struct Ipv4Header *ip = reinterpret_cast<struct Ipv4Header *>(eth + 1);
+
+    if (ip->ttl > 1) {
+      // Additional code for efficient checksum update for  changing TTL by an
+      // amount n
+      // TODO: Integration - add a check to identify an existing checksum?
+      // WARNING: though this code is efficient it can only update an already
+      // calculated checksum
+      // this code cannot calculate a checksum from scratch like ip_checksum.cc

```

Comment:
Just to make sure I remember what we decided with Barath here -- we decided not to update the checksum at all and let an external module re-compute the checksum before the packet leaves the system? The comment is a little unclear. Maybe just say: "Current design choice: implement a checksum at a downstream module instead of updating here. In the future, if we decided to have checksumming at each module, efficient code for checksum update (from RFC FOOBAR) is below."